softlayer/resource_softlayer_virtual_guest.go

```diff
@@ -470,12 +470,62 @@ func resourceSoftLayerVirtualGuestCreate(d *schema.ResourceData, meta interface{
 
 	log.Println("[INFO] Creating virtual machine")
 
-	guest, err := service.CreateObject(&opts)
-	if err != nil {
-		return fmt.Errorf("Error creating virtual guest: %s", err)
+	var id int
+
+	if opts.BlockDevices != nil && opts.BlockDeviceTemplateGroup != nil {
+		bd := *opts.BlockDeviceTemplateGroup
+		opts.BlockDeviceTemplateGroup = nil
+
+		// Find a temporary OS reference code that will be used to generate a template
```

Comment:
You can use something like "UBUNTU_LATEST" which is a code that won't be going away any time soon.